python/pyarrow/tests/test_array.py

```diff
@@ -3635,12 +3651,139 @@ def test_list_view_from_arrays(list_array_type):
     assert array.sizes.to_pylist() == sizes
 
 
-@pytest.mark.parametrize(('list_array_type'),
-                         [pa.ListViewArray, pa.LargeListViewArray])
-def test_list_view_flatten(list_array_type):
+@pytest.mark.parametrize(('list_array_type', 'list_type_factory'),
+                         [(pa.ListViewArray, pa.list_view),
+                          (pa.LargeListViewArray, pa.large_list_view)])
+def test_list_view_from_arrays_fails(list_array_type, list_type_factory):
+    values = [1, 2]
+    offsets = [0, 1, None]
+    sizes = [1, 1, 0]
+    mask = pa.array([False, False, True])
+
+    # Ambiguous to specify both validity map and offsets or sizes with nulls
+    with pytest.raises(pa.lib.ArrowInvalid):
+        list_array_type.from_arrays(offsets, sizes, values, mask=mask)
+
+    offsets = [0, 1, 1]
+    array = list_array_type.from_arrays(offsets, sizes, values, mask=mask)
+    array_slice = array[1:]
+
+    # List offsets and sizes must not be slices if a validity map is specified
+    with pytest.raises(pa.lib.ArrowInvalid):
+        list_array_type.from_arrays(
+            array_slice.offsets, array_slice.sizes,
+            array_slice.values, mask=array_slice.is_null())
+
+
+@pytest.mark.parametrize(('list_array_type', 'list_type_factory'), (
+    (pa.ListViewArray, pa.list_view),
+    (pa.LargeListViewArray, pa.large_list_view)
+))
+@pytest.mark.parametrize("arr", (
+    [None, [0]],
+    [None, [0, None], [0]],
+    [[0], [1]],
+))
+def test_list_view_from_factory(
+    list_array_type, list_type_factory, arr
+):
+    arr = pa.array(arr, type=list_type_factory(pa.int8()))
+    reconstructed_arr = list_array_type.from_arrays(
+        arr.offsets, arr.sizes, arr.values, mask=arr.is_null())
```

Comment:
```suggestion
    result = pa.array(arr, type=list_type_factory(pa.int8()))
    assert result.to_pylist() == arr
```

Maybe something like this instead? Because right now it doesn't really guarantee it was constructed correctly

Reply:
Although this test might also not add really value compared to what is already tested in test_convert_builtin.py?

Reply:
I will remove it, since I also think it is a redundant test. I duplicated this test from list/largelist unit tests unnecessarily.